### PR TITLE
metashell 4.0.0

### PR DESCRIPTION
**Pull request by r0mai**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/metashell.rb [hunk 1]
```diff
@@ -37,6 +37,7 @@ def install
       template <class T> struct add_const { using type = const T; };
       add_const<int>::type
     EOS
-    assert_match /const int/, shell_output("cat #{testpath}/test.hpp | #{bin}/metashell -H")
+    output = pipe_output("#{bin}/metashell -H", "cat #{testpath}/test.hpp, 0)
```
Comment (javian) [inline Formula/metashell.rb]: Looks like you forgot a trailing `"` =)
20:24:01 ==> FAILED
20:24:01 Error: Invalid formula: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/metashell.rb
20:24:01 metashell: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/metashell.rb:41: syntax error, unexpected tIDENTIFIER, expecting ')'
20:24:01     assert_match "const int", output
20:24:01                        ^
20:24:01 /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/metashell.rb:43: unterminated string meets end of file
20:24:01 /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/metashell.rb:43: syntax error, unexpected end-of-input, expecting keyword_end

Comment (r0mai) [reply Formula/metashell.rb]: @DomT4 I added the `"`. I hope you don't mind.

**Comment by DomT4** [comment]
The test rewrite broke in a suitably baffling way so I just reverted it locally until I have time to prod it a bit more.

Merged in https://github.com/Homebrew/homebrew-core/commit/63b384db242d46474c882904666c33c79978c911. Thank you for your contribution to Homebrew @r0mai; we appreciate it! :smiley_cat:

**Comment by r0mai** [comment]
Thanks @DomT4!

**Pull request closed without merging**